.github/workflows/ci.yml

```diff
@@ -0,0 +1,575 @@
+          RUSTFLAGS:                   -Clink-arg=-z -Clink-arg=nostart-stop-gc
+          # Since we run the tests with `--all-features` this implies the feature
+          # `ink-fuzz-tests` as well -- i.e. the fuzz tests are run.
+          # There's no way to disable a single feature while enabling all features
+          # at the same time, hence we use this workaround.
+          QUICKCHECK_TESTS:            0
+        run: |
+          cargo test --verbose --all-features --no-fail-fast --workspace
+          cargo test --verbose --all-features --no-fail-fast --workspace --doc
+
+  docs:
+    runs-on: ubuntu-latest
+    needs: [check, fmt, clippy, clippy-examples, dylint, spellcheck]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-directories: ${{ env.CARGO_TARGET_DIR }}
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Create docs
+        env:
+          RUSTDOCFLAGS:                  -Dwarnings
+        run: |
+          # All crate docs currently need to be built separately. The reason
+          # is that `smart-bench-macro` is a dependency now in a number of places.
+          # This crate uses e.g. `ink_metadata`, but in its published form. So if
+          # e.g. the `-p ink_metadata` is added to the `ink_lang` command  this
+          # results in the cargo failure "multiple packages with same spec, ambiguous".
+          cargo doc --no-deps --all-features -p ink_env
+          cargo doc --no-deps --all-features -p ink_storage
+          cargo doc --no-deps --all-features -p ink_storage_traits
+          cargo doc --no-deps --all-features -p ink_primitives
+          cargo doc --no-deps --all-features -p ink_prelude
+          cargo doc --no-deps --all-features -p ink
+          cargo doc --no-deps --all-features -p ink_macro
+          cargo doc --no-deps --all-features -p ink_ir
+          cargo doc --no-deps --all-features -p ink_codegen
+          cargo doc --no-deps --all-features -p ink_metadata
+          cargo doc --no-deps --all-features -p ink_e2e
+          cargo doc --no-deps --all-features -p ink_e2e_macro
+          mv ${CARGO_TARGET_DIR}/doc ./crate-docs
+          # FIXME: remove me after CI image gets nonroot
+          chown -R nonroot:nonroot ./crate-docs
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.job }}_${{ github.sha }}-doc
+          path: ./crate-docs
+          retention-days: 7
+
+  codecov:
+    runs-on: ubuntu-latest
+    needs: [check, fmt, clippy, clippy-examples, dylint, spellcheck]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
```

Comment:
```suggestion
      image: ${{ env.IMAGE }}
```

Reply:
I think that in this context, environment variables are not being handled